src/psyclone/psyGen.py

```diff
@@ -2203,7 +2208,7 @@ def field(self):
     @property
     def dag_name(self):
         ''' Return the name to use in a dag for this node'''
-        name = ("haloexchange({0})_".format(self._field.name) +
+        name = ("{0}({1})_".format(self._dag_name, self._field.name) +
```

Comment:
Might as well be consistent and use format for all three:
```suggestion
        name = ("{0}({1})_{2}".format(self._dag_name, self._field.name, self.position)
```

Reply:
Agreed.